### PR TITLE
Kotlin extension functions

### PR DESCRIPTION
**Pull request by dehydrogenaza**
Hey! I added some extension functions to make JColor more convenient to use in Kotlin. Let me know if you would like to integrate them to JColor (I'll take care of tests & docs), or perhaps I should just keep them as a fork.

Usage:
```
val kotlinFormat = AnsiFormat(Attribute.RED_TEXT(), Attribute.WHITE_BACK())

println("infix notation..." colorize kotlinFormat)

println("...or as an extension of String".colorize(Attribute.BLUE_BACK()))
```

### CODE REVIEW

**Comment by dialex** [comment]
Hey, thanks for your interest in contributing 😉 

This is a tough one, I don't have experience with Kotlin. Those extension functions you mentioned, can those be added and co-exist with a Java project? Or do we have to fork it and create a separate library?

My concerns are, after we merge your changes:
- Will it change the current API/contract of this JColor library?
- Will a Java dev be able to continue using this library in their Java env?
- Will a Kotlin dev be able to start using this library in their Kotlin env?

**Comment by dehydrogenaza** [comment]
Hey, so first thing to note is that I'm pretty new to this ("this" being programming in general), so I might be wrong about how things work ;-)

However, to my best knowledge:

- The Java methods aren't affected in any way.
- Kotlin compiles to normal JVM-compatible bytecode. Kotlin functions can be called in a normal fashion from Java (they would become static methods of KAnsi class), but there wouldn't really be any reason to do so in this case (reason below).
- A Kotlin dev can already use JColor without issue, using standard syntax. Extension/infix functions are just *syntactic sugar* that use Kotlin syntax to make code more readable. They are not *necessary* in any way.

Basically, I started by creating one of those function for my own personal convenience, but then decided to add a couple more just in case anyone else would like to have them out-of-the-box. To give you an example, here's how you'd normally call Java code from Kotlin:
`println(Ansi.colorize("normal invocation", myFormat))`

With extension function (my personal preference):
`println("extension function".colorize(myFormat))`

Or some people prefer the equivalent infix notation:
`println("infix function" colorize myFormat)`

**Comment by dialex** [comment]
I'll try to have a look this weekend. Sorry, I'm having limited availability lately. But thanks for the PR and the clarification. And your honesty too, hehe, all of us have been "new to this" at some point, and you are contributing to Open Source, so you are already on step ahead 😉

**Comment by dialex** [comment]
Sorry I took so long (2 months, oof). I decided to not merge this PR.

> **Subjective rationale:** This is a Java lib and I don't want to mix Kotlin. This PR adds a list of Kotlin specific dependencies, which goes against my principle of keeping dependencies to a bare minimum. I have 0 experience with Kotlin so I would not be able to properly test or maintain this in the future. I would basically close my eyes, press merger, and hope it never creates any issue.

But thank you for creating this PR. Anyone else that faces the same limitation can manually add your code to their project.

Happy Easter! 🐰

**Comment by dehydrogenaza** [comment]
Fully understood & agreed. BTW I found out about this project because it's used in the "official" Kotlin tutorial/course by Jetbrains Academy. Just wanted to let you know because I think that's pretty cool ;-)

**Comment by dialex** [comment]
Thanks for understanding 🙇 
Wow, really?? I had no idea! That makes me so happy and fulfilled, to know that my lib is being used and loved by other people. Do you have any link or screenshot to that course? I would love to take a screenshot of their usage of my library.

**Comment by dehydrogenaza** [comment]
Sure. This is from a free course so I hope I'm not breaking any laws by publishing this screenshot. 😅 I think there is some more that's paywalled, too.
Link: [https://hyperskill.org/learn/step/5076](https://hyperskill.org/learn/step/5076) (might require login)
Screenshot:
jcdp-hyperskill
And BTW I just noticed that they're still using JCDP (old version, too), I should probably let them know that there's newer stuff available.

**Comment by dialex** [comment]
Thank you so much for sending the screenshot! Feels good to see the lib being used out there :)

Sure, you can let them know. The new version will require some rework of their code though. Thanks!
